core/commands/active.go

```diff
@@ -35,48 +36,39 @@ Lists running and recently run commands.
 		"clear":    clearInactiveCmd,
 		"set-time": setRequestClearCmd,
 	},
-	Marshalers: map[cmds.EncodingType]cmds.Marshaler{
-		cmds.Text: func(res cmds.Response) (io.Reader, error) {
-			v, err := unwrapOutput(res.Output())
-			if err != nil {
-				return nil, err
-			}
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, out *[]*cmds.ReqLogEntry) error {
+			verbose, _ := req.Options[verboseOptionName].(bool)
 
-			out, ok := v.(*[]*cmds.ReqLogEntry)
-			if !ok {
-				return nil, e.TypeErr(out, v)
-			}
 			buf := new(bytes.Buffer)
```

Comment:
We should probably just write directly to `w` instead of the intermediate buffer.